src/thread.rs

```diff
@@ -406,7 +478,15 @@ impl<'mir, 'tcx: 'mir> ThreadManager<'mir, 'tcx> {
         }
         // We have not found a thread to execute.
         if self.threads.iter().all(|thread| thread.state == ThreadState::Terminated) {
-            unreachable!();
+            unreachable!("all threads terminated without the main thread terminating?!");
+        } else if let Some(sleep_time) =
+            self.timeout_callbacks.values().map(|info| info.call_time.get_wait_time()).min()
```

Comment:
Looks like this is re-doing the same computation above? Since that's non-trivial and accesses external state (the kernel clock), can we query just once and re-use the result here?